mne_bids/path.py

```diff
@@ -895,15 +908,25 @@ def get_entities_from_fname(fname):
     idx_key = 0
     for match in re.finditer(param_regex, op.basename(fname)):
         key, value = match.groups()
-        if key not in fname_vals:
-            raise KeyError(f'Unexpected entity "{key}" found in '
-                           f'filename "{fname}"')
-        if fname_vals.index(key) < idx_key:
-            raise ValueError(f'Entities in filename not ordered correctly.'
-                             f' "{key}" should have occurred earlier in the '
-                             f'filename "{fname}"')
-        idx_key = fname_vals.index(key)
-        params[ALLOWED_PATH_ENTITIES_SHORT[key]] = value
+
+        if on_error in ('raise', 'warn'):
+            if key not in fname_vals:
+                msg = f'Unexpected entity "{key}" found in '\
+                      f'filename "{fname}"'
```

Comment:
I think in other places we wrap long strings like these in parens instead of using backslashes. Could you please do the same here?